compiler/rustc_privacy/src/lib.rs

```diff
@@ -77,54 +79,54 @@ impl<'tcx, V> DefIdVisitorSkeleton<'_, 'tcx, V>
 where
     V: DefIdVisitor<'tcx> + ?Sized,
 {
-    fn visit_trait(&mut self, trait_ref: TraitRef<'tcx>) -> bool {
+    fn visit_trait(&mut self, trait_ref: TraitRef<'tcx>) -> ControlFlow<(), ()> {
         let TraitRef { def_id, substs } = trait_ref;
-        self.def_id_visitor.visit_def_id(def_id, "trait", &trait_ref.print_only_trait_path())
-            || (!self.def_id_visitor.shallow() && substs.visit_with(self))
+        if self.def_id_visitor.visit_def_id(def_id, "trait", &trait_ref.print_only_trait_path()) {
```

Comment:
should `visit_def_id` potentionally also return `ControlFlow`? :thinking:

Reply:
Yes, I think it should